contracts/swap/swap.go

```diff
@@ -41,7 +41,7 @@ type Backend interface {
 
 // SimpleSwap interface defines the simple swap's exposed methods
 type SimpleSwap interface {
-	Deploy(auth *bind.TransactOpts, backend bind.ContractBackend, owner common.Address) (common.Address, *types.Transaction, error)
+	Deploy(auth *bind.TransactOpts, backend bind.ContractBackend, owner common.Address, harddepositTimeout *big.Int) (common.Address, *types.Transaction, error)
```

Comment:
this would then have a `time.Duration` signature and you convert to `big.Int` before calling the smart contract

Reply:
Done.